File: No_2763/Sum of Imbalance Numbers of All Subarrays.cpp
class Solution {
public:
    int sumImbalanceNumbers(vector<int>& nums) {
        int ret = 0;
        for (int i = 0; i < nums.size(); i++) {
            unordered_set<int> hset;
            int count = -1;
            for (int j = i; j < nums.size(); j++) {
                if (hset.find(nums[j]) == hset.end()) {
                    if (hset.find(nums[j] - 1) == hset.end() && hset.find(nums[j] + 1) == hset.end())
                        count++;
                    else if (hset.find(nums[j] - 1) != hset.end() && hset.find(nums[j] + 1) != hset.end())
                        count--;
                }
                
                hset.insert(nums[j]);
                ret += count;
            }
        }
        
        return ret;
    }
};

File: No_2763/Sum of Imbalance Numbers of All Subarrays.py
class Solution:
    def sumImbalanceNumbers(self, nums: List[int]) -> int:
        n = len(nums)
        ret = 0
        for i in range(n):
            s = set()
            count = -1
            for num in nums[i:]:
                if num not in s:
                    if (num - 1) not in s and (num + 1) not in s:
                        count += 1
                    elif (num - 1) in s and (num + 1) in s:
                        count -= 1
                
                s.add(num)
                ret += count
                
        return ret
                        